core/src/main/scala/org/apache/spark/scheduler/TaskResult.scala

```diff
@@ -46,6 +46,8 @@ class DirectTaskResult[T](var valueBytes: ByteBuffer, var accumUpdates: Map[Long
 
   def this() = this(null.asInstanceOf[ByteBuffer], null, null)
 
+  override def toString: String = valueBytes.toString + " " + accumUpdates + " " + metrics
```

Comment:
This representation isn't really useful. It doesn't even tell you the task ID. I would just revert this change since it's unrelated and doesn't add much value.